### ISSUE TITLE
Possible bug when a write back request occurs to HBM

### ISSUE THREAD

**Issue by allencho1222** [gpgpu-sim/gpgpu-sim_distribution#180]
In general, HBM uses 128-bits data bus with 2 burst length (BL) (Assume we are not using pseudo channel mode). As a result, 32 bytes data can be gathered within a cycle (DDR).

In the current GPGPU-sim model, a write-back request is generated if an L2 cache line is dirty and the cache line has to be evicted. The address of the mem_fetch object is aligned to 128 bytes, which is the size of one cache line.

The problem here is that if we use the HBM memory model, one write back request (128 bytes aligned in the current GPGPU-sim design) has to be divided into four write back requests since HBM uses 128-bits data bus with 2 BL (32 bytes). However, regardless of the memory model, GPGPU-sim just generates one 128 bytes write back request and send it to the memory.

Moreover, if the sector cache and HBM are used, we don't have to write back 128 bytes cache line. The write requests can be generated only for dirty sectors (e.g., 2 write requests (32 bytes per request) can be generated for 2 sectors out of 4 sectors). However, I cannot make sure that this is the correct design for the write-back scheme in the real GPU machine.

To summarize, when we use HBM model, DRAM simulation is not correctly done since 128 bytes data (generated from the write-back request) are written in a single cycle given the HBM timing model (e.g., tCCDS = 1).

Please leave a comment if I'm wrong.

**Comment by allencho1222** [comment]
In dram.cc, I found that, depending on the requested data size, the commands are issued.
